### PR TITLE
ensure cuFile JNI library is loaded before any use [skip ci]

### PR DESCRIPTION
**Pull request by rongou**
Followup to #8053 to make sure `libcufilejni.so` is always loaded before creating `CuFileBuffer` or `CuFileHandle`.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/CuFile.java [hunk 1]
```diff
@@ -34,29 +34,24 @@
...
+    try {
+      NativeDepsLoader.loadNativeDeps(new String[]{"cufilejni"});
+      driver = new CuFileDriver();
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        driver.close();
+      }));
+      return true;
+    } catch (Throwable t) {
+      log.error("Could not load cuFile jni library...", t);
+      return false;
```
[the remark this answers is not in the document]
Comment (rongou) [reply java/src/main/java/ai/rapids/cudf/CuFile.java]: Done.

File: java/src/main/java/ai/rapids/cudf/CuFileBuffer.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
  * Represents a cuFile buffer.
  */
 public final class CuFileBuffer extends BaseDeviceMemoryBuffer {
+  @SuppressWarnings("unused")
+  private static final boolean CU_FILE_LIBRARY_LOADED = CuFile.libraryLoaded();
```
[the remark this answers is not in the document]
Comment (rongou) [reply java/src/main/java/ai/rapids/cudf/CuFileBuffer.java]: Done.

File: java/src/main/java/ai/rapids/cudf/CuFile.java [hunk 3]
```diff
@@ -56,6 +56,7 @@ private static synchronized void initialize() {
         initialized = true;
       } catch (Throwable t) {
         log.error("Could not load cuFile jni library...", t);
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/CuFile.java:58]: This is a log-then-throw antipattern, we should just throw and let the catching code decide whether to log.  This is very likely to be doubly-logged in practice.

Comment (rongou) [reply java/src/main/java/ai/rapids/cudf/CuFile.java]: Done.

**Review by jlowe — APPROVED** [review]

**Comment by rongou** [comment]
@gpucibot merge

**Pull request merged**